### PR TITLE
Add a basic skin line-up test scene

### PR DESCRIPTION
**Pull request by peppy**
Current elements are largely placeholder and will require further iteration.

- [x] Depends on https://github.com/ppy/osu/pull/5502.
- [x] Depends on https://github.com/ppy/osu/pull/5503.

### CODE REVIEW

File: osu.iOS.props [hunk 1]
```diff
@@ -35,7 +35,7 @@
     <DefineConstants></DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <CodesignKey>iPhone Distribution</CodesignKey>
+    <CodesignKey>iPhone Developer</CodesignKey>
```
Comment (smoogipoo) [inline osu.iOS.props]: Intended?

File: osu.Game.Rulesets.Osu.Tests/SkinnableTestScene.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        public void SetContents(Func<Drawable> creationFunction)
+        {
+            Cell(0).Child = new LocalSkinOverrideContainer(null) { RelativeSizeAxes = Axes.Both }.WithChild(creationFunction());
+            Cell(1).Child = new LocalSkinOverrideContainer(metricsSkin) { RelativeSizeAxes = Axes.Both }.WithChild(creationFunction());
+            Cell(2).Child = new LocalSkinOverrideContainer(defaultSkin) { RelativeSizeAxes = Axes.Both }.WithChild(creationFunction());
+            Cell(3).Child = new LocalSkinOverrideContainer(specialSkin) { RelativeSizeAxes = Axes.Both }.WithChild(creationFunction());
+        }
+
+        protected SkinnableTestScene()
```
Comment (smoogipoo) [inline osu.Game.Rulesets.Osu.Tests/SkinnableTestScene.cs]: ctor at the very bottom?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.iOS.props (hunk 1 above)
Comment (peppy) [reply osu.iOS.props]: very much no.

**Comment by smoogipoo** [comment]
What's a "metrics" skin? Is this some random skin that you've copied into our source? If so it needs to be removed and replaced with assets that are original.

**Comment by peppy** [comment]
You mean "special"?

**Comment by smoogipoo** [comment]
Ah yes, bottom right is special.

**Review by smoogipoo — APPROVED** [review]

**Comment by jorolf** [comment]
I'm currently trying to make some tests for #5497, and I was going to add some ``hitXX`` textures. How should they look? ``default`` should probably have the textures from stable but what about the others?

**Comment by smoogipoo** [comment]
As far as I understood it, the metrics skin has the exact outlines and expected paddings for elements. E.g. for hitcircles it has the dots on the side that indicate the padding, and the crosshair in the centre that should be visible. The special skin is just a user skin of some kind.

@peppy give direction plz

**Comment by peppy** [comment]
Metrics should be a png crafted to make it easy for the test viewer to know that the metrics are correct. In the case of `hitcircle`/`hitcircleoverlay`, I showed the relationship between them. For hit explosions, there's less to worry. Maybe a circle that starts precisely the same size as the hitcircle itself would be good (so scale animations are very apparent). Probably just a border, not filled.

**Pull request merged**